File: moreapps/src/main/java/com/rocky/moreapps/MoreAppsLifecycleListener.java
package com.rocky.moreapps;

public interface MoreAppsLifecycleListener {
    /**
     * AppCompatActivity or Fragment onStart LifeCycle Method
     */
    void onStart();

    /**
     * AppCompatActivity or Fragment onStop LifeCycle Method
     */
    void onStop();

    /**
     * updater dialog is showing, stop other works
     */
    void showingDialog();

    /**
     * on completing all processes regarding updater, continue other work
     */
    void onComplete();
}

File: moreapps/src/main/java/io/github/raghavsatyadev/moreapps/MoreAppsLifecycleListener.kt
package io.github.raghavsatyadev.moreapps

interface MoreAppsLifecycleListener {
    /**
     * AppCompatActivity or Fragment onStart LifeCycle Method
     */
    fun onStart()

    /**
     * AppCompatActivity or Fragment onStop LifeCycle Method
     */
    fun onStop()

    /**
     * updater dialog is showing, stop other works
     */
    fun showingDialog()

    /**
     * on completing all processes regarding updater, continue other work
     */
    fun onComplete()
}